### ISSUE TITLE
sudo wp cli update thoughts

### ISSUE THREAD

**Issue by cgsmith** [wp-cli/wp-cli#3181]
`sudo wp cli update --nightly` is required if the phar is placed in `/usr/local/bin`. This isn't encouraged without adding a root flag and the message of "YIKES" should provide a clearer message. For example, since `wp` in under `/user/local/bin` this command fails:

```
$ sudo -u chris -i -- wp cli update --nightly
Error: /usr/local/bin is not writable by current user
```

This requires the end user to run as root then ` sudo chown chris:chris /usr/local/bin/wp` to the file.

Composer is able to self-update without having to do these steps - I did not look into how. I am opening this issue to help start a discussion on making this process easier. What is everyone's thoughts?

**Comment by cgsmith** [comment]
Just a side note, i didn't know i had to match nightly until digging through #1310 and a [comment](https://github.com/wp-cli/wp-cli/issues/1631#issuecomment-224112973) on #1631 

A normal user should not have to dig that far to find out they need to update to nightly. Let me know how i can help with this as well.

**Comment by danielbachhuber** [comment]
> What is everyone's thoughts?

You should make `/usr/local/bin` writable by the user you'd like to update WP-CLI with.

**Comment by danielbachhuber** [comment]
> Just a side note, i didn't know i had to match nightly until digging through #1310 and a comment on #1631

Why do you mean by "had to match nightly"?

**Comment by cgsmith** [comment]
In order to get around wp-cli errors I had to update it to the nightly build. This was not immediately clear while searching, I thought wp-cli was just broken.

It appears that `/usr/local/bin` is assign root:root by default. I was also wrong as `composer self-update` does require elevated permissions.

My thoughts are to allow a user to run `sudo wp update` without a warning message - if they are running sudo they should understand the risks.

**Comment by danielbachhuber** [comment]
> My thoughts are to allow a user to run sudo wp update without a warning message - if they are running sudo they should understand the risks.

But they probably don't, so the decision was made long ago to introduce the `--allow-root` flag. It's not something I really want to debate again at depth.

**Comment by cgsmith** [comment]
Should the documentation be updated to explain that `/usr/local/bin` should be writable by the user?

**Comment by danielbachhuber** [comment]
> Should the documentation be updated to explain that /usr/local/bin should be writable by the user?

Could. I was thinking about putting a mention of `--allow-root` in "Common issues and their fixes" (#3119), but we could add some mention in the [updating docs](http://wp-cli.org/docs/installing/#updating-wp-cli).

Want to contribute a PR? [Here's an intro](http://wp-cli.org/docs/contributing/#improving-our-documentation)

**Comment by cconversion** [comment]
this continues to bug me, for simplicity workaround i have reolved the following line to add to your .zshrc file (or equivalent):
`alias wp="sudo wp"`
